Behaviour:
  Body:

    Toggles: # toggles für das Kopfverhalten
      # toggles für den Fieldie
      Fieldie:
          # lässt den Defender warten bis der ball bei sich in der nähe ist oder niemand sonst auf dem Feld ist um bis
          # zum nächsten set ein striker zu werden
          oneTimeDefender: True

          # Tries to self positionate at start of the geame
          trySelfPositioning : False

          # Cares about the Kickoff, for example wait for enemies
          careAboutKickoff : False

          # versuche mit dem goalie abzuklären ob wir aufs eigene tor schießen
          goalieNoOwnGoal: True

          # versucht sich vor dem Schießen zum Tor auszurichten
          alignToGoal : False

          # hackiges align to goal von dr wm2014
          hackAlign: False

          # benutzt den Sidekick im Spiel
          useSideKickInGame : False

          # use one time defender instead real defender for kick off strategy
          kickOffOneTimeDefender: False

          # use the dynamic kick made by the practica
          useDynamicKick: False

          # Track Ball and Goal while running to Ball
          trackBoth: False

          # Go away from Ball if opponent team has penalty/freekick
          penaltykick_go_away: False


      # Toggles für den Goalie
      Goalie:
          # Aktiviert das Relocate, das der Torwart sich im Tor richtig dreht, wenn er sich beim Hinwerfen umgedreht hat
          relocateTurn : True
          # lässt den goalie zum Fieldie werden
          goalieGoFieldie: False
          # ob der Goalie zum Ball hingeht, wenn er in der Nähe liegenbleibt
          goalieGoToBall: False
          # Ob der Goalie im Tor nach Rechts und links Läuft
          walkInGoal : False
          # ob der goalie beide arme hochnimmt und nich nur einen. Hotfix for to slow arm changings
          useBothArms: False
          # if the goalie tries to turn itself to the right direction, facing the opposite goal
          checkDirection: False

      # toggles für elfmeter schütze
      PenaltyFieldie:
          #lässt ihn direkt ohne weiteres verhalten schießen
          directPenaltyKick: True

          # benutze ein besonderes pathfinding für den penaltykick
          useSpecialPathfinding: True

    Fieldie:
      # Abstand bei dem der Roboter direkt vorm Ball steht um den ball beim kick zu erreichen(Fieldie)
      kickDistance:  0.2
      # Mindestabstand bei dem der Roboter direkt vorm Ball steht und den Ball kicken kann (Fieldie)
      minKickDistance:  0

      # Mindestwinkel zum tor, bevor er einen Sidekick anstatt einem Normalen Kick probiert
      minVSideKick: 1.0

      # Seitlicher versatz bei dem er sich noch sitlich zum Ball ausrichten soll
      kickAlign: 0.1

      # Der absolut Wert des angularen Wertes des Walkings mit dem er sich dreht wenn er nach dem Ball sucht.
      searchingTurnAngularAbsolute: 8

      # The maximum time the robot tries to align to the goal
      maxGoalAlignTime: 15

      # The time the robots does not try to align to the goal after triggering the stopping
      stopGoalAlignDuration: 10

      # The max distance to the enemy goal for using a hard kick
      maxGoalHardKickDistance: 3.0

      # Distance when he starts to track only the ball
      focusBallDistance: 0.8


      Defender:
          # distanz zum ball bei der der defender direkt zum normalen spieler wird
          goStrikerRange: 1.5
          # zeit die der defender am nächsten zum ball sein muss bis er denkt das er alleine ist
          goStrikerTime: 1.0
          # die wie viel letzte ball entfernenungen betrachtet werden um zu entscheiden ob der defender zum striker wird
          ballHistoryLenght: 20
          # Wie viele Trues in der History sein müssen zum striekr zu werden
          requiredNumberTrues: 18
          # Zeit die der Defender am Anfang nichts macht um nicht zum striker zu werden
          waitAtStart: 10
          # Zeit die der Defender Fieldie sein darf
          maxFieldieTime: 60

      KickOff:
          # Distance to ball, below the robot is the striker, above it is supporter, in m
          roleDecisionDistance: 0.5
          # time after the supporter ignores the kick off and starts normal behaviour
          ignoreKickOffTime: 20
          # Time to decide if a recived strategy was from a kick off before
          strategyOutdateTime: 30

    Goalie:
    ### Go To Ball ###
      # geschwindigkeit, bei der der Torwart denkt, dass der Ball liegt und er zu ihm geht
      goToBallVelocity: 0.1
      # u distanz bei der er zum Ball läuft
      goToBallu : 0.45

    ### Sideward walking in goal ###
      # Winkel ab dem es sich nicht lohnt seitlich im Tor zu laufen in Degree
      sidewardMoveAngleThreshold: 20
      # Distanz ab der wir uns seitwärts ausrichten um den ball zu blocken
      sidewardMoveUThreshold: 3.0
      # maximal movement to the the side as percent of the goal width
      maxSidewardMovement: 0.8

    ### Turn to right direction in goal ###
      # error in the difference to the wanted facing angle of the robot towards the oppsosite goal
      directionAngleThreshold: 30

    ### Throw ###
      # v distance to decide between right, middle and left throw
      vThrowDirectionDistance: 0.1
      # u estimation distance between befor throwing
      uThrowThreshold: 0.5

    # elfemter schütze
    PenaltyFieldie:
        # verhaltensrichtung festlegen
        # -1 zufällig entscheiden lassen
        # 0 direct kick
        # 1 etwas nach links
        # 2 etwas nach rechts
        # 3 etwas gerade aus
        penaltyDirection: -1

    OneTimeKicker:
        #time in seconds till the one time kicker is reseted to its original duty
        resetTime: 20




    Common:
      # Time after which the robot assumes he lost the ball
      maxBallTime: 3

      # how long a kick off is valid in s
      kickOffValidTime: 30

      # how long a drop ball is valid in s
      dropBallValidTime: 30

      # threshold at which the current position of the robot is counted as right
      positioningThreshold: 0.5

      Walking:
          # Forward-Backward
          SLOW_BACKWARD: -3
          MEDIUM_BACKWARD: -5
          FAST_BACKWARD: -7

          SLOW_FORWARD: 3
          MEDIUM_FORWARD: 5
          FAST_FORWARD: 7

          # Left-Right
          SLOW_SIDEWARDS_LEFT: 1
          MEDIUM_SIDEWARDS_LEFT: 3
          FAST_SIDEWARDS_LEFT: 5

          SLOW_SIDEWARDS_RIGHT: -1
          MEDIUM_SIDEWARDS_RIGHT: -3
          FAST_SIDEWARDS_RIGHT: -5

          # Angular Turn Left, Right
          SLOW_ANGULAR_LEFT: 2
          MEDIUM_ANGULAR_LEFT: 6
          FAST_ANGULAR_LEFT: 12

          SLOW_ANGULAR_RIGHT: -2
          MEDIUM_ANGULAR_RIGHT: -6
          FAST_ANGULAR_RIGHT: -12

      Positions:
          help: |
              All positions are given as per cent of
              half field length, half field width so
              that (0,0) is the center point, (1,1)
              is the corner to the left of the opponent
              goal and (-1,-1) is the right own corner

                       _____________________
              1    M  |          |          |  O
              |    Y  |_ -x, y   |   x, y  _|  P
              |       | |        |        | |  P
              0    G  | |       ( )       | |  G
              |    O  |_|        |        |_|  O
              |    A  |  -x,-y   |   x,-y   |  A
             -1    L  |__________|__________|  L

                     -1----------0----------1


          goalie: (-0.95, 0)
          teamPlayer: (-0.1, 0)
          defender: (-0.5, 0.2)
          center: (-0.7, -0.2)

      Field:
          length: 9
          width: 6
